controllers/capabilities/openapi_controller.go

```diff
@@ -261,13 +300,108 @@ func (r *OpenAPIReconciler) checkProductSynced(resource *capabilitiesv1beta1.Ope
 func (r *OpenAPIReconciler) readOpenAPI(resource *capabilitiesv1beta1.OpenAPI) (*openapi3.T, error) {
 	// OpenAPIRef is oneOf by CRD openapiV3 validation
 	if resource.Spec.OpenAPIRef.SecretRef != nil {
+		// Label the OAS source secret and OpenAPI so the secret can be watched by the openapi_controller
+		err := r.labelOpenAPISecretAndCR(resource)
+		if err != nil {
+			return nil, err
+		}
+
 		return r.readOpenAPISecret(resource)
 	}
 
 	// Must be URL
+	// Since the OpenAPI CR is URL ref, annotate it with a requeue interval so we can watch the OAS source for changes
+	err := r.annotateOpenAPICR(resource)
+	if err != nil {
+		return nil, err
+	}
+
 	return r.readOpenAPIFromURL(resource)
 }
 
+func (r *OpenAPIReconciler) labelOpenAPISecretAndCR(openAPICR *capabilitiesv1beta1.OpenAPI) error {
+	fieldErrors := field.ErrorList{}
+	specFldPath := field.NewPath("spec")
+	openapiRefFldPath := specFldPath.Child("openapiRef")
+	secretRefFldPath := openapiRefFldPath.Child("secretRef")
+
+	oasSourceSecret := &corev1.Secret{}
+	objectKey := types.NamespacedName{Name: openAPICR.Spec.OpenAPIRef.SecretRef.Name, Namespace: openAPICR.Spec.OpenAPIRef.SecretRef.Namespace}
+
+	// Read the OAS source secret
+	if err := r.Client().Get(r.Context(), objectKey, oasSourceSecret); err != nil {
+		if errors.IsNotFound(err) {
+			fieldErrors = append(fieldErrors, field.Invalid(secretRefFldPath, openAPICR.Spec.OpenAPIRef.SecretRef, "Secret not found"))
+			return &helper.SpecFieldError{
+				ErrorType:      helper.InvalidError,
+				FieldErrorList: fieldErrors,
+			}
+		}
+		// Unexpected error
+		return err
+	}
+
+	// Add label to OAS source secret so it can be watched
+	if oasSourceSecret.ObjectMeta.Labels == nil {
+		oasSourceSecret.ObjectMeta.Labels = map[string]string{}
+	}
+	oasSourceSecret.ObjectMeta.Labels[oasSecretLabelSelectorKey] = oasSecretLabelSelectorValue
+	if err := r.Client().Update(r.Context(), oasSourceSecret); err != nil {
+		return err
+	}
+
+	// Re-fetch the OpenAPI CR in case it's been modified
+	objectKey = types.NamespacedName{Name: openAPICR.Name, Namespace: openAPICR.Namespace}
+	if err := r.Client().Get(r.Context(), objectKey, openAPICR); err != nil {
+		return err
+	}
+
+	// Add label to OpenAPI CR with source secret's UID
+	if openAPICR.ObjectMeta.Labels == nil {
+		openAPICR.ObjectMeta.Labels = map[string]string{}
+	}
+	openAPICR.ObjectMeta.Labels[openAPISecretRefLabelKey] = string(oasSourceSecret.GetUID())
+	if err := r.Client().Update(r.Context(), openAPICR); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *OpenAPIReconciler) annotateOpenAPICR(openAPICR *capabilitiesv1beta1.OpenAPI) error {
```

Comment:
something small but would it be better to move that func to openapi_types?

Reply:
I removed this function altogether because we decided to remove functionality to let the user specify the OAS URL scrape interval.